plugins/language_st.lua

```diff
@@ -0,0 +1,127 @@
+-- mod-version:3
+local syntax = require "core.syntax"
+
+-- Language syntax references
+-- https://pdhonline.com/courses/e334/e334content.pdf
+
+syntax.add {
+  name = "PLC Structured Text IEC 61131-3",
+  files = { "%.stx?$", "%.iecst$" },
+  comment = "//",
+  block_comment = { "%(%*", "%*%)" },
+  patterns = {
+    { pattern = "//.*",                                     type = "comment"  }, -- Single-line comment
+    { pattern = { "%(%*", "%*%)" },                         type = "comment"  }, -- Multi-line comment
+    { pattern = { '"', '"', '\\' },                         type = "string"   }, -- String, quotation marks
+    { pattern = { "'", "'", '\\' },                         type = "string"   }, -- String, apices
+    { regex   = "\\s\\w+\\#[0-9]+[m|s]?",                   type = "number"   }, -- Time/Date formats
+    { regex   = "(?>[^(\\W]+)(?=\\()",                      type = "function" }, -- Function
+    { pattern = "[%a_][%w_]*",                              type = "symbol"   }, -- Symbols
+    { regex   = "^\\s*[A-Z]+_[A-Z]*\\s?\\;?\\s?",           type = "keyword"  }, -- Keyword
```

Comment:
```suggestion
    { regex   = [[^\s*[A-Z]+_[A-Z]*\s*;?]],                 type = "keyword"  }, -- Keyword
```
Is this intended to be available only at the start of the line?
Is the underscore in the first pattern supposed to be there, outside the character class?

Reply:
Yes and yes. It needs to match things like `TIME_AND_DATE` and `DATE`, at the beginning of the line.

Reply:
That regex is not doing that, it's only matching one underscore, and it's only matching when the underscore is present.

Reply:
Actually I think this regex pattern isn't even needed, because it should do what the symbol lua pattern (`[%a_][%w_]*`) does.
And the keywords are highlighted even without this regex pattern.
I don't remember why it was there honestly, it's been too long.